diffpy/pdfmorph/pdfmorph_api.py

```diff
@@ -0,0 +1,200 @@
+
+default_config = dict(scale=None, stretch=None, smear=None,
+                      baselineslope=None, qdamp=None)
+
+# map of operation dict
+# TODO: include morphing on psize
+morph_step_dict = dict(scale=morphs.MorphScale,
+                       stretch=morphs.MorphStretch,
+                       smear=[morphs.MorphXtalPDFtoRDF,
+                              morphs.MorphSmear,
+                              morphs.MorphXtalRDFtoPDF],
+                       qdamp=morphs.MorphResolutionDamping)
+
+def pdfmorph(xobj, yobj, xref, yref, rmin=None, rmax=None, rstep=None,
+             pearson=False, addpearson=False, fixed_operations=None,
+             refine=True, verbose=True, **kwargs):
+    """function to perfom PDF morphing.
+
+    Parameters
+    ----------
+    xobj : numpy.array
+        An array of objective x values, i.e., those will be manipulated by
+        morphing.
+    yobj : numpy.array
+        An array of objective y values, i.e., those will be manipulated by
+        morphing.
+    xref : numpy.array
+        An array of reference x values, i.e., those will be kept constant by
+        morphing.
+    yobj : numpy.array
+        An array of reference y values, i.e., those will be kept constant by
+        morphing.
+    rmin : float, optional
+        A value to specify lower r-limit of morph operations.
+    rmax : float, optional
+        A value to specify upper r-limit of morph operations.
+    rstep : float, optional
+        A value to specify rstep of morph operations.
+    pearson: Bool, optional
+        Option to include Pearson coefficient as a minimizing target 
+        during morphing. Default to False.
+    addpearson: Bool, optional
+        Option to include **both** Pearson coefficient and Rw as
+        minimizing targets during morphing. Default to False.
+    fixed_operations : list, optional
+        A list of string specifying operations will be keep fixed during
+        morphing. Default is None.
+    refine : bool, optional
+        Option to execute the minimization step in morphing. If False,
+        the morphing will be applied with parameter values specified in
+        `morph_config`. Default to True.
+    verbose : bool, optional
+        Option to print full result after morph. Default to True.
+    kwargs : dict, optional
+        A dictionary with morph parameters as keys and initial
+        values of morph parameters as values. Currently supported morph
+        parparameters are:
+
+            - 'scale'
+            - 'stretch'
+            - 'smear'
+            - 'baselineslope'
+            - 'qdamp'
+
+    Returns
+    -------
+    chain : diffpy.pdfmorph.morphs.morphchain.MorphChain
+        instance of morph chain. ``xobj, yobj, xref, yref = chain.xyallout``
+        will return morphed results
+    rv_cfg : dict
+        dictionary of refined parameters
+    rw : float
+        normal rw value
+    pcc : float
+        pearson correlation coefficient
```

Comment:
if there are so many returned objects, it is maybe better to return a dictionary or namedtuple.
I would have tough time to remember if the 3rd returned value is rw or pcc.

Reply:
👍

Reply:
I have a preference for `dict` in this case.